lib/ruby_lsp/erb_document.rb

```diff
@@ -0,0 +1,103 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  class ERBDocument < Document
+    extend T::Sig
+
+    sig { override.returns(Prism::ParseResult) }
+    def parse
+      return @parse_result unless @needs_parsing
+
+      @needs_parsing = false
+      scanner = ERBScanner.new(@source)
+      scanner.scan
+      @parse_result = Prism.parse(scanner.ruby)
+    end
+
+    class ERBScanner
+      extend T::Sig
+
+      sig { returns(String) }
+      attr_reader :ruby, :html
+
+      sig { params(source: String).void }
+      def initialize(source)
+        @source = source
+        @html = T.let(+"", String)
+        @ruby = T.let(+"", String)
+        @current_pos = T.let(0, Integer)
+        @inside_ruby = T.let(false, T::Boolean)
+      end
+
+      sig { void }
+      def scan
+        while @current_pos < @source.length
+          scan_char
+          @current_pos += 1
+        end
+      end
+
+      private
+
+      sig { void }
+      def scan_char
```

Comment:
`<%==` is valid syntax, basically does `raw`/`html_safe`. Should that be considered here?

I'm not sure if these are supported via plain `erb` but it certainly works with `erubi`.

Reply:
Thanks, we'll look into that.

Are there any other ERB gems other than those 2? I couldn't see any when searching.

Reply:
Found https://rubygems.org/gems/erubis/ but last release was 2011.

Reply:
I'm not aware of any other (recent) other implementation myself. `erubi` is the one used by rails and while it technically does allow to switch it out to something else that doesn't seem to be documented/used anymore. Rails 5.2 removed the `erubis` handler

Reply:
This is a good catch 👍. Based on [this test case](https://github.com/ruby/erb/blob/5d1c5108b600786a03ac3f38fcbf3f6054294d4c/test/erb/test_erb.rb#L582-L606), I think it's not supported by `erb` but is an expected extension, so I think it makes sense to support it.